src/sdc11073/pysoap/soapclientpool.py

```diff
@@ -1,33 +1,39 @@
-from typing import Protocol, Callable, Any
-from .. import loghelper
+from __future__ import annotations
 
+from typing import TYPE_CHECKING, Any, Callable, Protocol
+
+from sdc11073 import loghelper
+
+if TYPE_CHECKING:
+    from .soapclient import SoapClientProtocol
 
 _SoapClientFactory = Callable[[str, list[str]], Any]
```

Comment:
return type could be `SoapClientProtocol` or not?
can also be moved to `TYPE_CHECKING`